Modules/Devices/src/Devices.Application/Infrastructure/Persistence/Repository/IOAuthClientsRepository.cs

```diff
@@ -1,14 +1,14 @@
 ﻿using Backbone.Modules.Devices.Domain.Entities;
-using OpenIddict.EntityFrameworkCore.Models;
+using Backbone.Modules.Devices.Domain.OpenIddict;
 
 namespace Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
 
 public interface IOAuthClientsRepository
 {
     Task<IEnumerable<OAuthClient>> FindAll(CancellationToken cancellationToken);
-    Task<OpenIddictEntityFrameworkCoreApplication> Find(string clientId, CancellationToken cancellationToken);
+    Task<CustomOpenIddictEntityFrameworkCoreApplication> Find(string clientId, CancellationToken cancellationToken);
```

Comment:
Why do you now start to return a CustomOpenIddictEntityFrameworkCoreApplication when we return an `OAuthClient` in the other methods?

Reply:
When we want to update a Client, we need the complete `CustomOpenIddictEntityFrameworkCoreApplication` object for the `_applicationManager.UpdateAsync` method.
We could technically change it so it returns an `OAuthClient`, but then we would need to call the `_applicationManager.FindByClientIdAsync` inside the `Update` and `ChangeClientSecret` methods to get the full object.

Reply:
Another alternative would be implementing your own tracking mechanism. So: in each `Find` method, save the `CustomOpenIddictEntityFrameworkCoreApplication`s in a list. After that you can map them to `OAuthClient`s and return them. 
Then, when `Update` is called, check if the list contains a `CustomOpenIddictEntityFrameworkCoreApplication`. For the unlikely case it doesn't contain one, you can read it using the application manager.